controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +247,62 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    /*
+        Conditional delete of the specified key with the specified version.
+        If the conditional delete fails it is ignored.
+     */
+    private CompletableFuture<Void> conditionalDeleteOfKey(String tableName, long requestId, TableSegmentKey key,
+                                                           TableSegmentKeyVersion keyVersion) {
+        // Compute the list of keys to be deleted.
+        List<TableSegmentKey> conditionalDeleteKeys = Collections.singletonList(TableSegmentKey.versioned(key.getKey(), keyVersion.getSegmentVersion()));
+        Supplier<String> errorMessage = () -> String.format("conditionalDeleteOfKey: table: %s", tableName);
+        // attempt to do a conditional delete of the keys.
+        return withRetries(() -> segmentHelper.removeTableKeys(tableName, conditionalDeleteKeys, authToken.get(), requestId), errorMessage, requestId)
+                .exceptionally(t -> {
+                    log.info(requestId, "Conditional remove table keys failed, ignoring this exception {}", t.getMessage());
+                    return null;
+                }).whenComplete((v, t) -> releaseKeys(conditionalDeleteKeys));
+    }
+
+    /**
+     * Method to get and conditionally update value for the specified key.
+     *
+     * This method fetches the latest version of the value for the specified key and applies the update function and
+     * updates the value.
+     *
+     * @param tableName Table Name.
+     * @param tableKey Key to update.
+     * @param updateFunction A function which updates the values.
+     * @param attemptCleanup A Predicate which decides if a cleanup should be invoked.
+     * @param requestId Request id.
+     * @return A future which completes when the update operation has completed. A conditional update failure will cause
+     * the future to complete exceptionally.
+     */
+    public CompletableFuture<Void> getAndUpdateValues(final String tableName, final String tableKey, final Function<TableSegmentEntry, TableSegmentEntry> updateFunction,
```

Comment:
nit: its updating a single entry - should we rename this method to getAndUpdateEntry instead?

Reply:
Yes, corrected it.